src/Importify/Cabal.hs

```diff
@@ -3,28 +3,102 @@
 
 module Importify.Cabal
        ( getLibs
+       , getExtensionMaps
        , readCabal
+       , moduleNameToPath
+       , TargetMap
+       , ExtensionsMap
        ) where
 
-import           Universum
+import           Universum                             hiding (fromString)
 
+import           Data.List                             (nub)
+import qualified Data.Map.Strict                       as Map
+import           Distribution.ModuleName               (ModuleName, fromString,
+                                                        toFilePath)
 import           Distribution.Package                  (Dependency (..), PackageName (..))
-import           Distribution.PackageDescription       (GenericPackageDescription (..),
-                                                        condTreeData, libBuildInfo,
-                                                        targetBuildDepends)
+import           Distribution.PackageDescription       (BuildInfo, Executable,
+                                                        GenericPackageDescription (..),
+                                                        Library, buildInfo,
+                                                        condExecutables, condTreeData,
+                                                        defaultExtensions, exeModules,
+                                                        libBuildInfo, libModules,
+                                                        modulePath, targetBuildDepends)
 import           Distribution.PackageDescription.Parse (readPackageDescription)
 import           Distribution.Verbosity                (normal)
+import           Language.Haskell.Extension            (Extension (..))
+
+type TargetMap = Map String String
+type ExtensionsMap = Map String [String]
 
 readCabal :: FilePath -> IO GenericPackageDescription
 readCabal = readPackageDescription normal
 
--- TODO: also collect for executables and make unique
 -- TODO: what about version bounds?
 getLibs :: GenericPackageDescription -> [String]
-getLibs GenericPackageDescription{..} =
-    maybe []
-          (map dependencyName . targetBuildDepends . libBuildInfo . condTreeData)
-          condLibrary
+getLibs = nub . concat . map (map dependencyName . targetBuildDepends) . getBuildInfos
+
+getBuildInfos :: GenericPackageDescription -> [BuildInfo]
+getBuildInfos GenericPackageDescription{..} =
+    maybe [] ((:[]) . libBuildInfo . condTreeData) condLibrary ++
+    map (buildInfo . condTreeData . snd) condExecutables
+
+getExtensionMaps :: GenericPackageDescription -> (TargetMap, ExtensionsMap)
+getExtensionMaps GenericPackageDescription{..} =
+    ( foldr Map.union Map.empty $ libTargetsMaps ++ exeTargetsMaps
```

Comment:
There's [`unions`](http://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Map-Lazy.html#v:unions) function to unite list of maps.